test/lib/y2firewall/dialogs/main_test.rb

```diff
@@ -54,12 +54,26 @@
 
   describe "#run" do
     let(:result) { :next }
+    let(:firewall_service) { instance_double("Yast2::SystemService", save: true) }
 
     before do
       allow_any_instance_of(CWM::Dialog).to receive(:run).and_return(result)
+      allow(firewall).to receive(:system_service).and_return(firewall_service)
     end
 
     context "when the user accepts the changes" do
+      it "writes the firewall configuration" do
+        expect(firewall).to receive(:write_only)
+
+        subject.run
+      end
+
+      it "updates the firewalld systemd service status" do
```

Comment:
I would write another test to be sure that the `save` method *is not called* when the user aborts.